File: Samples/VB.NET/CodeExamples/MessageComposerSample.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Inetlab.SMPP
Imports Inetlab.SMPP.Common
Imports Inetlab.SMPP.Logging
Imports Inetlab.SMPP.PDU

Namespace CodeSamples
	Public Class MessageComposerSample
		Private _log As ILog = LogManager.GetLogger(GetType(MessageComposerSample).FullName)
		'<EventsSample>
		Private ReadOnly _client As New SmppClient()
		Private ReadOnly _composer As New MessageComposer()

		Public Sub New()
			AddHandler _client.evDeliverSm, AddressOf client_evDeliverSm

			AddHandler _composer.evFullMessageReceived, AddressOf OnFullMessageReceived
			AddHandler _composer.evFullMessageTimeout, AddressOf OnFullMessageTimedout
		End Sub

		Private Sub client_evDeliverSm(ByVal sender As Object, ByVal data As DeliverSm)
			_composer.AddMessage(data)
		End Sub

		Private Sub OnFullMessageTimedout(ByVal sender As Object, ByVal args As MessageEventHandlerArgs)
			Dim pdu As DeliverSm = args.GetFirst(Of DeliverSm)()
			_log.Info(String.Format("Incomplete message received from {0}", pdu.SourceAddress))
		End Sub

		Private Sub OnFullMessageReceived(ByVal sender As Object, ByVal args As MessageEventHandlerArgs)
			Dim pdu As DeliverSm = args.GetFirst(Of DeliverSm)()
			_log.Info(String.Format("Full message received from {0}: {1}", pdu.SourceAddress, args.Text))
		End Sub
		'</EventsSample>


		'<InlineSample>

		Private Sub client_evDeliverSmInline(ByVal sender As Object, ByVal data As DeliverSm)
			_composer.AddMessage(data)
			If _composer.IsLastSegment(data) Then
			   Dim receivedText As String = _composer.GetFullMessage(data)
			End If
		End Sub

		'</InlineSample>
	End Class
End Namespace

File: Samples/CS/CodeSamples/MessageComposerSample.cs
using System;
using System.Collections.Generic;
using System.Text;
using Inetlab.SMPP;
using Inetlab.SMPP.Common;
using Inetlab.SMPP.Logging;
using Inetlab.SMPP.PDU;

namespace CodeSamples
{
    public class MessageComposerSample
    {
        private ILog _log = LogManager.GetLogger(typeof(MessageComposerSample).FullName);
        //<EventsSample>
        private readonly SmppClient _client = new SmppClient();
        private readonly MessageComposer _composer = new MessageComposer();

        public MessageComposerSample()
        {
            _client.evDeliverSm += client_evDeliverSm;

            _composer.evFullMessageReceived += OnFullMessageReceived;
            _composer.evFullMessageTimeout += OnFullMessageTimedout;
        }

        private void client_evDeliverSm(object sender, DeliverSm data)
        {
            _composer.AddMessage(data);
        }

        private void OnFullMessageTimedout(object sender, MessageEventHandlerArgs args)
        {
            DeliverSm pdu = args.GetFirst<DeliverSm>();
            _log.Info(string.Format("Incomplete message received from {0}", pdu.SourceAddress));
        }

        private void OnFullMessageReceived(object sender, MessageEventHandlerArgs args)
        {
            DeliverSm pdu = args.GetFirst<DeliverSm>();
            _log.Info(string.Format("Full message received from {0}: {1}", pdu.SourceAddress, args.Text));
        }
        //</EventsSample>


        //<InlineSample>

        private void client_evDeliverSmInline(object sender, DeliverSm data)
        {
            _composer.AddMessage(data);
            if (_composer.IsLastSegment(data))
            {
               string receivedText = _composer.GetFullMessage(data);
            }
        }

        //</InlineSample>
    }
}
